### PR TITLE
[RG-228] Add close_design tcl command

### PR DESCRIPTION
**Pull request by KochynVolodymyr**
### Motivate of the pull request
close_design command added.

### Describe the technical details
This command just close current design if it exists. If design doesn't exist, just do nothing.

 ### Which part of the code base require a change
 
 - [x] Library: compiler, newproject, foedagcore
 - [ ] Plug-in: <Specify the plugin name>
 - [ ] Engine
 - [ ] Documentation
 - [x] Regression tests
 - [ ] Continous Integration (CI) scripts

### CODE REVIEW

File: src/Compiler/Compiler.cpp [hunk 1]
```diff
@@ -112,6 +112,7 @@ void Compiler::Help(std::ostream* out) {
   (*out) << "   create_design <name> ?-type <project type>? : Creates a design "
             "with <name> name"
          << std::endl;
+  (*out) << "   close_design     : Close current design" << std::endl;
```
Comment (ravic-rs) [inline src/Compiler/Compiler.cpp:115]: Did we verify the behavior when close_design is used from the tcl console in the GUI ?

Comment (KochynVolodymyr) [reply src/Compiler/Compiler.cpp:115]: yes, I did it

File: src/MainWindow/main_window.cpp [hunk 2]
```diff
@@ -261,9 +261,9 @@ void MainWindow::openProjectDialog(const QString& dir) {
   if (!fileName.isEmpty()) openProject(fileName, false, false);
 }
 
-void MainWindow::closeProject() {
-  if (m_projectManager && m_projectManager->HasDesign() &&
-      confirmCloseProject()) {
+void MainWindow::closeProject(bool force) {
```
Comment (ravic-rs) [inline src/MainWindow/main_window.cpp:264]: Why do we need this flag ? close_design could just have been linked to closeProject.

Comment (KochynVolodymyr) [reply src/MainWindow/main_window.cpp:264]: because there is a message box. We don't need it in tcl command

**Review by ravic-rs — APPROVED** [review]

**Pull request merged**